### DISCUSSION TITLE
API test

### DISCUSSION THREAD

**Discussion by Adahai** [openucx/ucx#6677] [Q&A] [answered]
Hi, 
I want to write my own program to call UCP_TAG_SEND_NB and UCP_TAG_RECV_NB to realize a very simple function, which is to receive and send data(ucp_tag_send_nb and ucp_tag_recv_nb are ucx API )，How do I compile my program？

**Comment by petro-rudenko** [comment, marked as answer]
As described on [Compiling Applications with UCX](https://github.com/openucx/ucx/wiki/Compiling-Applications-with-UCX) you need to have built version of ucx, then just use something like this:
```
gcc -o ucp_hello_world.dynamic ucp_hello_world.c -L${UCX_LIB} -I${UCX_INC} -lucp -lucs -luct -pedantic -Werror -Wl,-rpath,${UCX_LIB}
```
